### ISSUE TITLE
Add an Trigger Notification Button on Pull Request View

### ISSUE THREAD

**Issue by christiangalsterer** [tomasbjerre/pull-request-notifier-for-stash#33]
For some use cases the notification should be send again, even if there was no event happening. E.g. a Jenkins build was triggered but due to a configuration error or technical problem the job shall be triggered again for this particular pull request. Therefore it would nice to add a pull a button "Trigger Notification on the overview page for each pull request similar to the "Trigger Build" of the [Stash Webhook to Jenkins]
(https://marketplace.atlassian.com/plugins/com.nerdwin15.stash-stash-webhook-jenkins) plugin.

**Comment by tomasbjerre** [comment]
I implemented that as a post-build-action in Jenkins with a small shell-script. The script comments the pull-request with something like "Looking good! Here is the log: http://... Trigger build again: http://...."

So I see the use case!

**Comment by christiangalsterer** [comment]
Thanks for the feedback and it some environments this certainly works. In larger environments and/or with PRs containing a lot comments "finding" the comment containing the comment with the link to trigger URL usability may be hampered.

As an implementation idea, maybe introduce a "Trigger Manually" event, so that the button only fires, when this is configured as a suitable event. 

Thanks for looking into this and evaluating if you see an option to support an dedicated button.
PS: If you can share the shell script anyways this would be highly appreciated.

**Comment by christiangalsterer** [comment]
Already found the script on your web page ;-)

**Comment by derekln1** [comment]
It would be nice to have a trigger build button.  But we are implementing manual triggers with comments.
When a COMMENTADDED event occurs and the comment contains the keyword for the particular build, it fires our Jenkins job:
- "verification_build" - triggers a build for the source commit of the pull request
- "pre_merge_build" - triggers a build for the resulting merge commit of the source and target branches

**Comment by fredrikand** [comment]
I agree with the original request, to have a button added in the pull-request view would be very nice!

I started implementing our CI setup with the "Stash webhook to Gerrit", and found it lacking in so many respects. I have been working 3 years with Gerrit on another assignment and the GerritTrigger integration Jenkins is way more advanced. The webhook approach does not include information about the pull request for example. Making it  very hard to automate things in Jenkins following the build.

Now when I found your plugin, things are looking brighter for Stash. I was about to throw it out the window in bringing in Gerrit instead.

Thank you for a great plugin!
